File: app/src/test/java/io/intrepid/contest/screens/contestcreation/describecontest/DescribeContestPresenterTest.java
package io.intrepid.contest.screens.contestcreation.describecontest;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.junit.MockitoJUnitRunner;

import io.intrepid.contest.models.Contest;
import io.intrepid.contest.testutils.TestPresenterConfiguration;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class DescribeContestPresenterTest {
    @Mock
    DescribeContestContract.View mockView;
    @Mock
    Contest.Builder mockContestBuilder;
    private DescribeContestPresenter describeContestPresenter;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);
        describeContestPresenter = new DescribeContestPresenter(mockView,
                                                                TestPresenterConfiguration.createTestConfiguration(),
                                                                mockContestBuilder);
    }

    @Test
    public void onViewCreatedShouldCauseViewToEnableNext() {
        describeContestPresenter.onViewCreated();
        verify(mockView).setNextEnabled(true);
    }

    @Test
    public void onNextClickShouldTriggerViewToShowNextScreen() {
        String VALID_TEXT = "Valid Contest Name";
        describeContestPresenter.onNextClicked(VALID_TEXT);
        verify(mockView).showNextScreen();
    }
}

File: app/src/test/java/io/intrepid/contest/screens/contestcreation/describecontest/DescribeContestPresenterTest.kt
package io.intrepid.contest.screens.contestcreation.describecontest

import io.intrepid.contest.models.Contest
import io.intrepid.contest.screens.contestcreation.describecontest.DescribeContestContract.View
import io.intrepid.contest.testutils.BasePresenterTest
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.Mockito.verify
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class DescribeContestPresenterTest : BasePresenterTest<DescribeContestPresenter>() {
    @Mock
    private lateinit var mockView: View
    @Mock
    private lateinit var mockContestBuilder: Contest.Builder

    @Before
    fun setup() {
        presenter = DescribeContestPresenter(mockView, testConfiguration, mockContestBuilder)
    }

    @Test
    fun onNextPageButtonClickedShouldTriggerViewToShowNextScreen() {
        val VALID_TEXT = "Valid Contest Name"
        presenter.onNextPageButtonClicked(VALID_TEXT)
        verify<View>(mockView).showNextScreen()
    }

    @Test
    fun isNextPageButtonEnabledShouldBeEnabled() {
        assertTrue(presenter.isNextPageButtonEnabled())
    }
}
